independent-projects/qute/core/src/main/java/io/quarkus/qute/Parser.java

```diff
@@ -1044,14 +1052,16 @@ private List<TemplateNode> readLines(List<List<TemplateNode>> lines, List<Templa
         return currentLine;
     }
 
-    private boolean isStandalone(List<TemplateNode> line) {
+    private boolean isStandalone(List<TemplateNode> line, List<String> dataNamespaceKeys) {
         boolean maybeStandalone = false;
         for (TemplateNode node : line) {
             if (node instanceof ExpressionNode) {
                 // Line contains an expression
                 return false;
-            } else if (node instanceof SectionNode || node instanceof ParameterDeclarationNode || node == BLOCK_NODE
-                    || node == COMMENT_NODE) {
+            } else if (node instanceof SectionNode
+                    || (node instanceof ParameterDeclarationNode
+                            && !dataNamespaceKeys.contains(((ParameterDeclarationNode) node).getKey()))
```

Comment:
What's this for? I mean `isStandalone()` is used to detect [standalone lines](https://quarkus.io/guides/qute-reference#removing-standalone-lines-from-the-template) and I see no connection with validation of `data:` expressions...

Reply:
we need parameter declarations for "data" namespace, but they are removed as standalone if there is no key associated with them (so we need to know there is an expression in data namespace associated with them).

Reply:
Ah, you're right. But that's more like an oversight/bug. We need to fix this.

Reply:
FYI https://github.com/quarkusio/quarkus/pull/26254